src/main.tsx

```diff
@@ -69,26 +89,65 @@ async function introduceWorkers(sharedWorker: SharedWorker) {
   ])
   console.log("posted to both workers")
 }
-introduceWorkers(sharedWorker)
 
-function setupSharedWorkerAndRepo() {
+function createRepo(sharedWorker: SharedWorker) {
   const repoNetworkChannel = new MessageChannel()
   sharedWorker.port.postMessage({ repoNetworkPort: repoNetworkChannel.port2 }, [
     repoNetworkChannel.port2,
   ])
 
-  const repo = new Repo({
+  return new Repo({
     network: [new MessageChannelNetworkAdapter(repoNetworkChannel.port1)],
     sharePolicy: (peerId) => peerId.includes("shared-worker"),
   })
+}
+
+async function setupRepoWithoutSharedWorker() {
```

Comment:
i don't love this part of the patch and it's not well-integrated with the rest of the system. would you split this into its own PR please?

some notes on this stuff:
- if there's no serviceworker we want to have a sharePolicy with the sync-server (otherwise new documents won't be pushed)
- we can do a better job of code reuse i suspect between the shared worker and here by making a separate file and putting this code into it
- does the serviceworker work when you do this? it's gonna block the main UI really badly.
- maybe we should just use a webworker in this situation?
- what breaks if you have two tabs writing at the same time. i don't think we do this right now but i'm pretty sure there are terrible data-corrupting race conditions lurking here.

Reply:
Sorry, that commit isn't supposed to be in here. I've started working on making blutack work without shared workers on a general fixes branch. I then rebased it onto this branch and later removed the shared workers logic but forgot to remove it here.